unlock-app/src/__tests__/services/web3Service.test.js

```diff
@@ -1454,5 +1482,22 @@ describe('Web3Service', () => {
         })
       })
     })
+
+    describe('_watchTransaction', () => {
+      it('should set a timeout based on half the block time', () => {
+        expect.assertions(1)
+        jest.useFakeTimers()
+        web3Service._watchTransaction('0x')
+        expect(setTimeout).toHaveBeenCalledTimes(1)
```

Comment:
How would you feel about making this assertion a bit more specific to ensure that the timing is close to our expected duration( half of the block time)?

Reply:
That's a good one. Adding it.